main/autogen/subclasses.cc

```diff
@@ -52,18 +65,8 @@ optional<Subclasses::Map> Subclasses::listAllSubclasses(core::Context ctx, Parse
             continue;
         }
 
-        // Get fully-qualified parent name as string
-        string parentName =
-            fmt::format("{}", fmt::map_join(ref.resolved.nameParts,
-                                            "::", [&ctx](const core::NameRef &nm) -> string { return nm.show(ctx); }));
-
-        // Add child class to the set identified by its parent
-        string childName =
-            fmt::format("{}", fmt::map_join(pf.showFullName(ctx, defn),
-                                            "::", [&ctx](const core::NameRef &nm) -> string { return nm.show(ctx); }));
-
-        auto &mapEntry = out[parentName];
-        mapEntry.entries.insert(make_pair(childName, defn.data(pf).type));
+        auto &mapEntry = out[ref.sym];
+        mapEntry.entries.insert({defn.data(pf), pf.file});
```

Comment:
So instead of storing `defn.data(pf)` which is a struct, I'd actually just store `defn.data(pf).sym` here. Save even more memory allocation and copying. In which case, you won't even need to store `pf.file`, because `defn.data(pf).sym.loc(ctx)` will be guaranteed to correspond to the right file. Basically it's then just a map of symbolRef -> list of symbolRefs, which should be even more lightweight.

Reply:
but then we would still have to store `pf` somewhere since it gets deleted after the workers are finished.

Reply:
Scratch that, I think this optimization is possible